src/js/types/generic.type.ts

```diff
@@ -23,3 +23,13 @@ export type Menu = GenericObject;
 export type Tooltip = GenericObject;
 
 export type Localization = GenericObject;
+
+export type SourceObj = {
+    type: string;
+    mimeType: string;
+};
+
+export type SeekRange = {
```

Comment:
this one i'm not sure where to house - `source` has a file that will get types but seek ranges are sort of spread throughout. I almost think this generic file might be the best place for em. thoughts?

Reply:
In DefaultProvider is fine. I don't like how the defaults are implemented now, but the idea is all providers extend it so it makes sense for types and _the_ provider interface be defined there.